warehouse/integrations/snowflake/snowflake.go

```diff
@@ -166,6 +168,18 @@ func New(conf *config.Config, log logger.Logger, stat stats.Stats) *Snowflake {
 	sf.logger = log.Child("integrations").Child("snowflake")
 	sf.stats = stat
 
+	loadTableStrategy := conf.GetString("Warehouse.snowflake.loadTableStrategy", loadTableStrategyMergeMode)
+	switch loadTableStrategy {
+	case loadTableStrategyMergeMode, loadTableStrategyAppendMode:
+		sf.config.loadTableStrategy = loadTableStrategy
+	default:
+		loadTableStrategy = loadTableStrategyMergeMode
+		sf.logger.Errorw(
+			"Received invalid loadTableStrategy, defaulting to MERGE mode",
+			lf.LoadTableStrategy, loadTableStrategy,
+		)
```

Comment:
Should we just return from here instead? Since it's set up only.

Reply:
I thought I'd fail gracefully, just log the error, set the default and carry on. If we re return from there then we don't setup `enableDeleteByJobs` and `slowQueryThreshold`. As an alternative I can make the constructor return an error and propagate it upstream.

Reply:
If we propagate the error upstream to the manager then it's going to end up in `uploadsToProcess` which leads to a `panic` so we could do that as well. Thoughts?

Reply:
I think we should we should add an error for the constructor.

Reply:
Although, we don't have a proper way to initialize these integrations. We are calling New but it does only partial initialization. Some are calling `Setup` while others rely on `Connect`. Or some are just calling functions like `TestConnection`, and `IsEmpty`.

Reply:
https://github.com/rudderlabs/rudder-server/pull/3745/commits/1a0fc96621e51c2ad15d5f9e90485650dad63ff5

Reply:
Do you mean the `warehouse/validations` package?

Reply:
> Do you mean the `warehouse/validations` package?

Yes.